milli/src/heed_codec/roaring_bitmap/bo_roaring_bitmap_codec.rs

```diff
@@ -7,9 +9,14 @@ pub struct BoRoaringBitmapCodec;
 impl heed::BytesDecode<'_> for BoRoaringBitmapCodec {
     type DItem = RoaringBitmap;
 
-    fn bytes_decode(mut bytes: &[u8]) -> Option<Self::DItem> {
+    fn bytes_decode(bytes: &[u8]) -> Option<Self::DItem> {
         let mut bitmap = RoaringBitmap::new();
-        while let Ok(integer) = bytes.read_u32::<NativeEndian>() {
+        let num_u32 = bytes.len() / size_of::<u32>();
+        for i in 0..num_u32 {
+            let start = i * size_of::<u32>();
+            let end = (i + 1) * size_of::<u32>();
+            let mut bytes = bytes.get(start..end)?;
+            let integer = bytes.read_u32::<NativeEndian>().ok()?;
```

Comment:
Wouldn't it be possible to use bytemuck here? Something like `bytemuck::try_cast_slice(bytes)` followed by a `slice.iter().copied().collect()` as [`RoaringBitmap` impl `FromIterator`](https://docs.rs/roaring/0.6.6/roaring/bitmap/struct.RoaringBitmap.html#impl-FromIterator%3Cu32%3E) ?